src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -100,9 +101,30 @@ private static OnnxTransform Create(IHostEnvironment env, ModelLoadContext ctx)
             if (!ctx.TryLoadBinaryStream("OnnxModel", r => modelBytes = r.ReadByteArray()))
                 throw env.ExceptDecode();
 
-            var inputColumn = ctx.LoadNonEmptyString();
-            var outputColumn = ctx.LoadNonEmptyString();
-            var args = new Arguments() { InputColumn = inputColumn, OutputColumn = outputColumn };
+            bool isMultiOutput = ctx.Header.ModelVerReadable > 0x00010001;
+
+            //var inputColumn = ctx.LoadNonEmptyString();
+            //var outputColumn = ctx.LoadNonEmptyString();
+
+            var numInputs = 1;
+            if (isMultiOutput)
+                numInputs = ctx.Reader.ReadInt32();
```

Comment:
var numInputs = isMultiOutput ?  ctx.Reader.ReadInt32() : 1;

Reply:
condensed

---
In reply to: [232842851](https://github.com/dotnet/machinelearning/pull/1586#discussion_r232842851) [](ancestors = 232842851)